sahi/model.py

```diff
@@ -1063,3 +1063,106 @@ def _create_object_prediction_list_from_original_predictions(
             object_prediction_list_per_image.append(object_prediction_list)
 
         self._object_prediction_list_per_image = object_prediction_list_per_image
+
+
+@check_requirements(["tensorflow", "tensorflow_hub"])
+class TensorflowHubDetectionModel(DetectionModel):
+    def load_model(self):
+        try:
+            import tensorflow_hub as hub
+
+        except ImportError:
+            raise ImportError(
+                "Please install tensorflow and tensorflow_hub via `pip install tensorflow' and 'pip install tensorflow_hub`"
+            )
+
+        self.model = hub.load(self.model_path)
+
+    def perform_inference(self, image: np.ndarray):
+        if self.image_size is not None:
+            from sahi.utils.tfhub import get_image, resize
+
+            img = get_image(image)
+            img = resize(img, self.image_size)
+
+            prediction_result = self.model(img)
+
+        else:
+            from sahi.utils.tfhub import get_image
+
+            img = get_image(image)
+
+            prediction_result = self.model(img)
+
+        self._original_predictions = prediction_result
+        self.img = get_image(image)
+
+        if self.category_mapping is None:
+            from sahi.utils.tfhub import COCO_CLASSES
+
+            category_mapping = {str(i): COCO_CLASSES[i] for i in range(len(COCO_CLASSES))}
+            self.category_mapping = category_mapping
+
+    @property
+    def num_categories(self):
+        num_categories = len(self.category_mapping)
+        return num_categories
+
+    @property
+    def has_mask(self):
+        return self.model.with_mask
+
+    @property
+    def category_names(self):
+        return list(self.category_mapping.values())
+
+    def _create_object_prediction_list_from_original_predictions(
+        self,
+        shift_amount_list: Optional[List[List[int]]] = [[0, 0]],
+        full_shape_list: Optional[List[List[int]]] = None,
+    ):
+        original_predictions = self._original_predictions
+
+        # compatilibty for sahi v0.8.20
+        if isinstance(shift_amount_list[0], int):
+            shift_amount_list = [shift_amount_list]
+        if full_shape_list is not None and isinstance(full_shape_list[0], int):
+            full_shape_list = [full_shape_list]
+
+        shift_amount = shift_amount_list[0]
+        full_shape = None if full_shape_list is None else full_shape_list[0]
+
+        boxes = original_predictions["detection_boxes"][0]
+        scores = original_predictions["detection_scores"][0]
+        category_ids = original_predictions["detection_classes"][0]
+
+        # create object_prediction_list
+        object_prediction_list = []
+        object_prediction_list_per_image = []
+
+        for i in range(min(boxes.shape[0], 100)):
+            if scores[i] >= self.confidence_threshold:
+                score = float(scores[i].numpy())
```

Comment:
her for loopunda numpy() ile cevirme islemi yapmak yerine for'a girmeden once bir kere yaparak hizi arttirabiliriz

```python
boxes = original_predictions["detection_boxes"][0].numpy()
scores = original_predictions["detection_scores"][0].numpy()
category_ids = original_predictions["detection_classes"][0].numpy()
```